pkg/odo/cli/url/describe.go

```diff
@@ -0,0 +1,112 @@
+const describeRecommendedCommandName = "describe"
+
+var describeExample = ktemplates.Examples(`  # Describe a URL
+%[1]s myurl
+`)
+
+// URLListOptions encapsulates the options for the odo url list command
+type URLDescribeOptions struct {
+	localConfigInfo  *config.LocalConfigInfo
+	componentContext string
+	url string
+	*genericclioptions.Context
+}
+
+// NewURLDescribeOptions creates a new URLCreateOptions instance
+func NewURLDescribeOptions() *URLDescribeOptions {
+	return &URLDescribeOptions{&config.LocalConfigInfo{}, "", "", &genericclioptions.Context{}}
+}
+
+// Complete completes URLDescribeOptions after they've been Listed
+func (o *URLDescribeOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
+	o.Context = genericclioptions.NewContext(cmd)
+	o.localConfigInfo, err = config.NewLocalConfigInfo(o.componentContext)
+	if err != nil {
+		return errors.Wrap(err, "failed intiating local config")
+	}
+	if len(args) == 1 {
+		o.url = args[0]
+	}
+	return
+}
+
+// Validate validates the URLDescribeOptions based on completed values
+func (o *URLDescribeOptions) Validate() (err error) {
+	return util.CheckOutputFlag(o.OutputFlag)
+}
+
+// Run contains the logic for the odo url list command
+func (o *URLDescribeOptions) Run() (err error) {
+
+	urls, err := url.List(o.Client, o.localConfigInfo, o.Component(), o.Application)
+	if err != nil {
+		return err
+	}
+	u, err := urls.Get(o.url)
+	if err != nil {
+		return err
+	}
+
+	if log.IsJSON() {
+		machineoutput.OutputSuccess(u)
+	} else {
+
+		tabWriterURL := tabwriter.NewWriter(os.Stdout, 5, 2, 3, ' ', tabwriter.TabIndent)
+		fmt.Fprintln(tabWriterURL, "NAME", "\t", "STATE", "\t", "URL", "\t", "PORT")
+
+		// are there changes between local and cluster states?
+		outOfSync := false
+		fmt.Fprintln(tabWriterURL, u.Name, "\t", u.Status.State, "\t", url.GetURLString(u.Spec.Protocol, u.Spec.Host), "\t", u.Spec.Port)
+		if u.Status.State != url.StateTypePushed {
+			outOfSync = true
+		}
+		tabWriterURL.Flush()
+		if outOfSync {
+			fmt.Fprintf(os.Stdout, "\n")
+			fmt.Fprintf(os.Stdout, "There are local changes. Please run 'odo push'.\n")
+		}
+	}
+
+	return
+}
+
+// NewCmdURLDescribe implements the odo url describe command.
+func NewCmdURLDescribe(name, fullName string) *cobra.Command {
+	o := NewURLDescribeOptions()
+	urlDescribeCmd := &cobra.Command{
+		Use:         name,
+		Short:       "Describe a URL",
+		Long:        `Describe a URL`,
+		Example:     fmt.Sprintf(describeExample, fullName),
+		Args:        cobra.RangeArgs(0, 1),
+		Annotations: map[string]string{"machineoutput": "json", "command": "url"},
+		Run: func(cmd *cobra.Command, args []string) {
+			genericclioptions.GenericRun(o, cmd, args)
+		},
+	}
+	urlDescribeCmd.SetUsageTemplate(util.CmdUsageTemplate)
+	genericclioptions.AddContextFlag(urlDescribeCmd, &o.componentContext)
+	completion.RegisterCommandFlagHandler(urlDescribeCmd, "context", completion.FileCompletionHandler)
```

Comment:
also add a command handler like we have in `url delete`

Reply:
will check that